File: Assets/Bundles/Shared/Resources/TS/UserInput/MobileJoystick.lua
-- Compiled with unity-ts v2.1.0-75
local Bin = require("Shared/TS/Util/Bin").Bin
local Signal = require("Shared/TS/Util/Signal").Signal
local MobileJoystickDriver = require("Shared/TS/UserInput/Drivers/MobileJoystickDriver").MobileJoystickDriver
local MobileJoystick
do
	MobileJoystick = setmetatable({}, {
		__tostring = function()
			return "MobileJoystick"
		end,
	})
	MobileJoystick.__index = MobileJoystick
	function MobileJoystick.new(...)
		local self = setmetatable({}, MobileJoystick)
		return self:constructor(...) or self
	end
	function MobileJoystick:constructor()
		self.bin = Bin.new()
		self.mobileJoystickDriver = MobileJoystickDriver:instance()
		self.Changed = Signal.new()
		self.bin:Add(self.Changed)
		self.bin:Add(self.mobileJoystickDriver.Changed:Proxy(self.Changed))
	end
	function MobileJoystick:IsVisible()
		return self.mobileJoystickDriver:IsVisible()
	end
	function MobileJoystick:SetVisible(visible)
		self.mobileJoystickDriver:SetVisible(visible)
	end
	function MobileJoystick:Destroy()
		self.bin:Destroy()
	end
end
return {
	MobileJoystick = MobileJoystick,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/@Easy/Core/Core~/src/Shared/UserInput/MobileJoystick.ts
import { Bin } from "Shared/Util/Bin";
import { Signal } from "Shared/Util/Signal";
import { MobileJoystickDriver } from "./Drivers/MobileJoystickDriver";

export class MobileJoystick {
	private readonly bin = new Bin();
	private readonly mobileJoystickDriver = MobileJoystickDriver.Instance();

	/** Fires when the position of the joystick changes, including when it's released. */
	public readonly changed = new Signal<[position: Vector3, phase: MobileJoystickPhase]>();

	/** Returns `true` if the mobile joystick is visible. */
	public IsVisible() {
		return this.mobileJoystickDriver.IsVisible();
	}

	/** Set the visibility of the mobile joystick. */
	public SetVisible(visible: boolean) {
		this.mobileJoystickDriver.SetVisible(visible);
	}

	constructor() {
		this.bin.Add(this.changed);
		this.bin.Add(this.mobileJoystickDriver.changed.Proxy(this.changed));
	}

	/** Cleans up the mobile joystick listener. */
	public Destroy() {
		this.bin.Destroy();
	}
}
